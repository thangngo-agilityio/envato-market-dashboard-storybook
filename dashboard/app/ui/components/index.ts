export { default as InputField } from './common/InputField';
export { default as FallbackErrorBoundary } from './FallbackErrorBoundary';
export { default as Navigation } from './common/Navigation';
export { default as Menu } from './common/Menu';
export { default as Logo } from './common/Logo';
export { default as ExpandSidebar } from './ExpandSidebar';
export { default as MiniSidebar } from './MiniSidebar';
export * from './Icons';
export * from './Images';
export { default as Indicator } from './Indicator';
export { default as Loading } from './common/Loading';
export { default as Avatar } from './common/Avatar';
export { default as Benefit } from './common/Benefit';
export { default as Divider } from './common/Divider';
export { default as IconButton } from './common/IconButton';
export { default as Dropdown } from './common/Dropdown';
export { default as Select } from './common/Select';
export { default as Pagination } from './common/Pagination';
export { default as Button } from './common/Button';
export { default as Table } from './common/Table';
export { default as CustomerNameCell } from './common/Table/Body/CustomerNameCell';
export { default as ProductNameCell } from './common/Table/Body/ProductNameCell';
export { default as ActionCell } from './common/Table/Body/ActionCell';
export { default as UserInfoCell } from './UsersTable/Cell/UserInfoCell';
export { default as StatusCell } from './UsersTable/Cell/StatusCell';
export { default as HeadCell } from './common/Table/HeadCell';
export { default as Selector } from './TransactionTable/Selector';
export { default as TransactionTable } from './TransactionTable';
export { default as SearchBar } from './common/SearchBar';
export { default as SwitchTheme } from './SwitchTheme';
export { default as CardPayment } from './CardPayment';
export { default as BoxChat } from './BoxChat';
export { default as TotalStatisticCard } from './TotalStatisticCard';
export { default as Fetching } from './Skeleton/Fetching';
export { default as TotalStatisticListSkeleton } from './Skeleton/TotalStatisticListSkeleton';
export { default as TotalStatisticList } from './TotalStatisticList';
export { default as RevenueFlow } from './RevenueFlow';
export { default as Efficiency } from './Efficiency';
export { default as TotalBalance } from './TotalBalance';
export { default as OverallBalance } from './OverallBalance';
export { default as UserCard } from './UserCard';
export { default as ItemSideBarSetting } from './ItemSideBarSetting';
export { default as FaqItem } from './Setting/Faq/FaqItem';
export { default as UpdateProfile } from './Profile';
export { default as AuthForm } from './AuthForm';
export { default as Modal } from './common/Modal';
export { default as Notification } from './common/Notification';
export { default as FallbackImage } from './FallbackImage';
export { default as ChatMember } from './ChatMember';
export { default as Conversation } from './Conversation';
export { default as BonusNotification } from './common/BonusNotification';
export { default as CardIssues } from './CustomerIssues/CardIssues';
export { default as CustomerIssues } from './CustomerIssues';
export { default as PinCode } from './common/PinCode';
export { default as Quill } from './Quill';
export { default as MessageAdmin } from './MessagesAdmin';
export { default as ListMessages } from './Conversation/ListMessage';
export { default as InputSendMessages } from './InputSendMessages';
export { default as PinCodeModal } from './PinCodeModal';
export { default as ProductsTable } from './ProductsTable';
export { default as UploadImages } from './UploadImages';
export { default as ProductForm } from './common/Table/Body/ProductForm';
export { default as RecentActivitiesTable } from './RecentActivitiesTable';
