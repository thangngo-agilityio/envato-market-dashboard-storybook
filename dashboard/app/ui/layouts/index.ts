export { default as SideBar } from './Sidebar';
export { default as Header } from './Header';
export { default as AuthHeader } from './AuthHeader';
export { default as AuthFooter } from './AuthFooter';
export { default as MainLayout } from './MainLayout';
