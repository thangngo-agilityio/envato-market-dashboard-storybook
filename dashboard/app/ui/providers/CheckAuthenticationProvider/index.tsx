'use client';

import { PRIVATE_ROUTES, PUBLIC_ROUTES, ROUTES } from '@/lib/constants';
import { TAuthStoreData, authStore } from '@/lib/stores';
import { redirect, usePathname } from 'next/navigation';
import { useEffect } from 'react';

type TValidateRoute = {
  id: number;
  path: string;
};

const CheckAuthenticationProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const pathname = usePathname();
  const user = authStore((state): TAuthStoreData['user'] => state.user);
  const isMatchPrivateRoute: boolean = PRIVATE_ROUTES.some(
    (route: TValidateRoute) =>
      pathname === ROUTES.ROOT || `/${route.path}` === pathname,
  );
  const isMatchPublicRoute: boolean = PUBLIC_ROUTES.some(
    (route: TValidateRoute) => `/${route.path}` === pathname,
  );

  useEffect(() => {
    if (isMatchPublicRoute && !!user) {
      return redirect(ROUTES.ROOT);
    }

    if (isMatchPrivateRoute && !user) {
      return redirect(ROUTES.LOGIN);
    }
  }, [isMatchPrivateRoute, isMatchPublicRoute, user]);

  return children;
};

export default CheckAuthenticationProvider;
