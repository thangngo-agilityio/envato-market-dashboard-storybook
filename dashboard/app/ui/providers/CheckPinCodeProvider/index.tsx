'use client';

import dynamic from 'next/dynamic';
import { useCallback, useMemo } from 'react';
import { useDisclosure, useToast } from '@chakra-ui/react';

import { SubmitHandler, useForm } from 'react-hook-form';

//Constants
import { ERROR_MESSAGES, STATUS, SUCCESS_MESSAGES } from '@/lib/constants';

// Hooks
import { useAuth, usePinCode } from '@/lib/hooks';

// Types
import { TPinCodeForm } from '@/lib/interfaces';

// Stores
import { TAuthStoreData, authStore } from '@/lib/stores';

// Utils
import { customToast } from '@/lib/utils';

const Modal = dynamic(() => import('@/ui/components/common/Modal'));
const PinCode = dynamic(() => import('@/ui/components/common/PinCode'));

const CheckPinCodeProvider = () => {
  const user = authStore((state): TAuthStoreData['user'] => state.user);

  const { isOpen: isPinCodeModalOpen, onClose: onClosePinCodeModal } =
    useDisclosure({ isOpen: true });

  const {
    control,
    handleSubmit,
    formState: { isValid, isSubmitting },
  } = useForm<TPinCodeForm>({
    defaultValues: {
      pinCode: '',
    },
  });

  const { setUser } = useAuth();

  const toast = useToast();

  const { handleSetPinCode } = usePinCode();

  const onSubmitPinCode: SubmitHandler<TPinCodeForm> = useCallback(
    (data) => {
      if (user) {
        data.userId = user.id;

        try {
          handleSetPinCode(data);

          setUser({ user: { ...user, pinCode: data.pinCode } });

          onClosePinCodeModal();

          toast(
            customToast(
              SUCCESS_MESSAGES.SET_PIN_CODE.title,
              SUCCESS_MESSAGES.SET_PIN_CODE.description,
              STATUS.SUCCESS,
            ),
          );
        } catch (error) {
          toast(
            customToast(
              ERROR_MESSAGES.SET_PIN_CODE.title,
              ERROR_MESSAGES.SET_PIN_CODE.description,
              STATUS.ERROR,
            ),
          );
        }
      }
    },
    [handleSetPinCode, onClosePinCodeModal, setUser, toast, user],
  );

  const pinCodeModalBody = useMemo(
    () => (
      <PinCode
        control={control}
        isDisabled={!isValid || isSubmitting}
        onSubmit={handleSubmit(onSubmitPinCode)}
        onClose={onClosePinCodeModal}
      />
    ),
    [
      control,
      handleSubmit,
      isSubmitting,
      isValid,
      onClosePinCodeModal,
      onSubmitPinCode,
    ],
  );

  return (
    isPinCodeModalOpen && (
      <Modal
        title="Please set the PIN code to your account"
        isOpen={isPinCodeModalOpen}
        onClose={onClosePinCodeModal}
        body={pinCodeModalBody}
      />
    )
  );
};

export default CheckPinCodeProvider;
