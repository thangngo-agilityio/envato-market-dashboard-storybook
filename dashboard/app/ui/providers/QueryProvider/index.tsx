'use client';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { memo, ReactNode } from 'react';
import { useState } from 'react';

// Constants
import { DEFAULT_STALE_TIME } from '@/lib/constants';

type TQueryProvider = {
  children: ReactNode;
};

const QueryProviderComponent = ({ children }: TQueryProvider) => {
  const [queryClient] = useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            staleTime: DEFAULT_STALE_TIME,
            refetchOnWindowFocus: false,
          },
        },
      }),
  );

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};

const QueryProvider = memo(QueryProviderComponent);

export default QueryProvider;
