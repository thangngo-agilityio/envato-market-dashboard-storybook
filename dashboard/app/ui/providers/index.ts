export { default as ChakraProvider } from './ChakraUIProvider';
export { default as QueryProvider } from './QueryProvider';
export { default as CheckAuthenticationProvider } from './CheckAuthenticationProvider';
export { default as CheckPinCodeProvider } from './CheckPinCodeProvider';
