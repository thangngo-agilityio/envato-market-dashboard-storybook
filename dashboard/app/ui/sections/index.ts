export { default as UsersSection } from './UsersSection';
export { default as SettingsSection } from './SettingsSection';
export { default as SupportsSection } from './SupportSection';
export { default as ForgotPasswordSection } from './ForgotPassword';
export { default as AccountSection } from './AccountSection';
export { default as DashBoardSection } from './DashBoardSection';
export { default as MyWalletSection } from './MyWalletSection';
export { default as TransactionSection } from './TransactionSection';
export { default as HistorySection } from './HistorySection';
export { default as ProductsSection } from './ProductsSection';
export { default as RecentActivitiesSection } from './RecentActivitiesSection';
