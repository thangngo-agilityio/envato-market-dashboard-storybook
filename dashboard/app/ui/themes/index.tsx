// Libs
import { extendTheme, Tooltip } from '@chakra-ui/react';

// Bases
import {
  radii,
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  letterSpacings,
  colors,
  sizes,
  breakpoints,
} from './bases';

// Components
import {
  Tag,
  Button,
  Checkbox,
  Drawer,
  FormError,
  Heading,
  Input,
  Text,
  Table,
  Badge,
  Skeleton,
  Spinner,
  Link,
  Select,
} from './components';

export const configThemes = {
  ...extendTheme({
    semanticTokens: {
      radii,
      space,
      fonts,
      fontSizes,
      fontWeights,
      lineHeights,
      letterSpacings,
      colors,
      sizes,
    },
    breakpoints,
    components: {
      Tag,
      Text,
      Button,
      Checkbox,
      Heading,
      Input,
      FormError,
      Drawer,
      Table,
      Badge,
      Skeleton,
      Spinner,
      Link,
      Menu: Select,
    },
  }),

  styles: {
    global: {
      'html, body': {
        fontFamily: 'primary',
      },
    },
  },
  initialColorMode: 'system',
  useSystemColorMode: true,
};

// Override the default properties of Tooltip component
Tooltip.defaultProps = { ...Tooltip.defaultProps, openDelay: 400 };
