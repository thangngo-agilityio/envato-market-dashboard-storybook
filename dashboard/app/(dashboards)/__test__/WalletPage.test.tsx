import { QueryProvider } from '@/ui/providers';

import { useSearchParams } from 'next/navigation';
import MyWallets from '../my-wallets/page';

jest.mock('react-intersection-observer');

describe('Dashboard render', () => {
  beforeAll(async () => {
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn(),
      forEach: jest.fn(),
    });
  });

  test('Should render match with snapshot.', async () => {
    const { container } = render(await MyWallets(), { wrapper: QueryProvider });
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
