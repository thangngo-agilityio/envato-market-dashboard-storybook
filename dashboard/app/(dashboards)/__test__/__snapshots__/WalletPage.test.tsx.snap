// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-et9gjv"
  >
    <div
      class="css-zd4of5"
    >
      <div
        class="css-6iskwu"
      />
    </div>
    <div
      class="css-gca1gs"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
